examples/history_vehicles_replacement_for_imitation_learning.py

```diff
@@ -99,8 +100,13 @@ def main(
     assert traffic_history_provider
 
     scenarios_iterator = Scenario.scenario_variations(scenarios, [])
-    for scenario in scenarios_iterator:
+    total_scenario_variations = Scenario.discover_traffic_history_scenarios_count(
+        scenarios
+    )
+    for _ in range(total_scenario_variations):
+        scenario = next(scenarios_iterator)
```

Comment:
This can just be shortened to using the iterator via `for`.
```suggestion
    for scenario in scenarios_iterator:
```

Reply:
`scenarios_iterator` is cyclic. If we iterate over it in a for loop, we will go into infinite loop. Thats why we had to workaround finding the total number of scenario variations.